### DISCUSSION TITLE
[RFC / Discussion] Spreadsheet features where it shines

### DISCUSSION THREAD

**Discussion by DVGY** [iddan/react-spreadsheet#342] [Q&A] [answered]
Nice work !!! 

I have worked extensively with react-data-grid and other JavaScript spreadsheets. I have few suggestions for you, these are just my experience and I am not imposing. 

Most of the JavaScript Spreadsheets are good to show data to users, but when it comes to **editing** they just don't work or lot of code need to be written , even after writing lot of code some of the functionality are not achievable 😄 

Comparison of few features that provides natively good UX and timesaving in spreadsheet.

| Features  | UX in spreadsheet | UX in react form |
| ------------- | ------------- | ---------------------- |
| Budgeting / Cost Estimations | Great | Not that great, lot of clicks for a user if he is coming from excel background  |
| Grouping Items like excel | Great | Great  | 
| Gantt Chart | Great | Great (required additional library installation) |

It would be great if this library in future can accommodate editing, cell formulas, related cells, grouping and etc.
PS: If a user decides to buy a licenses for spreadsheet, then most of the features works smoothly

**Comment by iddan** [comment, marked as answer]
Thanks for the detailed report @DVGY. I'm not sure which component and license are you referring to.
React Spreadsheet has no paid license, it's completely open source.
As for features:
 - **Editing** works as closely as possible to Google Sheets / Microsoft Excel
 - **Cell formulas** work as expected (including formula dependencies and circular dependency)
 - **Related cells** - not sure what do you mean by that
 - **Grouping** is something we haven't gotten into yet but I would love to review PRs relating to that.

**Reply by DVGY** [reply]
I meant license for paid spreadsheets like handsontable ag grid and etc.

**Reply by iddan** [reply]
Oh, you mean the other libraries out there require a license for these features to work correctly?

**Reply by DVGY** [reply]
Yesh
